src/Tribe/API.php

```diff
@@ -602,5 +642,222 @@ public static function get_and_flatten_event_meta( $event_id ) {
+						esc_html__( 'An event having a post meta value of `%s` meridian is not valid. Make sure it is either `am` or `pm`, or remove it entirely if using the 24-hour format.', 'the-events-calendar' ),
+						$value
+					);
+
+					return new WP_Error( 'invalid-tribe-events-meridian-meta-value', $error_message );
+				}
+			}
+
+			return $new_value;
+		}
+
+		/**
+		 * Sanitize a string to be used as an event hour post meta value: 1-12 if `Event{Start|End}Meridian`
+		 * is also passed, else 0-23.
+		 *
+		 * Use this prior to sending value to the database.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $value       The post meta value to be checked, such as '07'.
+		 * @param bool   $twelve_hour If false, allow 0-23. If true, only allow 1-12.
+		 *
+		 * @return string|WP_Error
+		 */
+		public static function sanitize_hour_meta_value( $value, $twelve_hour = false ) {
+			$value = (string) $value;
+
+			if ( ! is_numeric( $value ) ) {
+				$error_message = sprintf(
+					esc_html__( 'An event having a post meta value of `%s` hour (am/pm) is not valid. Make sure it is from 1 to 12.', 'the-events-calendar' ),
+					$value
+				);
+
+				return new WP_Error( 'non-numeric-tribe-events-hour-meta-value', $error_message );
+			}
+
+			$new_value = absint( $value );
+
+			if ( $twelve_hour ) {
+				if (
+					1 > $new_value
+					|| 12 < $new_value
+				) {
+					$error_message = sprintf(
+						esc_html__( 'An event having a post meta value of `%s` hour (12-hour) is not valid. Make sure it is from 1 to 12.', 'the-events-calendar' ),
+						$value
+					);
+
+					return new WP_Error( 'invalid-tribe-events-12-hour-meta-value', $error_message );
+				}
+			} else {
+				if ( 23 < $new_value ) {
+					$error_message = sprintf(
+						esc_html__( 'An event having a post meta value of `%s` hour (24-hour) is not valid. Make sure it is from 0 to 23.', 'the-events-calendar' ),
+						$value
+					);
+
+					return new WP_Error( 'invalid-tribe-events-24-hour-meta-value', $error_message );
+				}
+			}
+
+			$new_value = str_pad( $new_value, 2, '0', STR_PAD_LEFT );
+
+			return $new_value;
+		}
+
+		/**
+		 * Sanitize a string to be used as an event minute post meta value: 0-59.
+		 *
+		 * Use this prior to sending value to the database.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $value The post meta value to be checked, such as '30'.
+		 *
+		 * @return string|WP_Error
+		 */
+		public static function sanitize_minute_meta_value( $value ) {
+			$value = (string) $value;
+
+			if ( '' === trim( $value ) ) {
+				return trim( $value );
+			}
+
+			if ( ! is_numeric( $value ) ) {
+				$error_message = sprintf(
+					esc_html__( 'An event having a post meta value of `%s` minutes is not valid. Make sure it is from 0 to 59.', 'the-events-calendar' ),
+					$value
+				);
+
+				return new WP_Error( 'non-numeric-tribe-events-minutes-meta-value', $error_message );
+			}
+
+			$new_value = absint( $value );
+
+			if ( 59 < $new_value ) {
```

Comment:
I think by flipping the condition provides a more easier way to understand this conditional

```php
if ( $new_value > 59 )
```

- *`$new_value` is greater than 59*

Instead of 

- *`59 is greater than $new_value`*

Reply:
https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions says to do it the way I did it, correct?

Reply:
That's correct as specified on Codex, main reason for this is to avoid confusion on assigning values like this: 

```
if ( $a = 12 ) {}
``` 

However I think on most of services sides we are moving away from this one for couple of reasons: 

1. Readability as suggested on first comment.
2. We have code reviews in place which give us an opportunity to catch for that possible scenario.
3. We are using strict comparison most of the time ( `===` `!===` ) and for `>`, `<` there's more hard to fall into the problem that "tries to solve" the problem faced by WP Yoda style

Reply:
okie dokie

Reply:
Per Slack conversation, I should change all to variable on the left.